### PR TITLE
Reduced model with F/T sensors

### PR DESCRIPTION
**Pull request by diegoferigo**

### CODE REVIEW

**Comment by traversaro** [comment]
The macOS build is failing because the `sensors` library's compilation units are being still compiled as C++98 (we are switching one library at the time using the `idyntree_enable_cxx11()` macro). Adding the `idyntree_enable_cxx11()` macro in the sensors' CMakeLists.txt should solve the issue. 
Other than that, there is something else keeping this PR in `[WIP]` state?

**Comment by diegoferigo** [comment]
Ok thanks, I'll enable C++11 to solve macOS build.

Still `[WIP]`, the commit for the time being is just blind code, I still have to write a test for the reduced model with sensors.

**Comment by traversaro** [comment]
Ok!

**Comment by DanielePucci** [comment]
Any update on this @diegoferigo ?

**Comment by diegoferigo** [comment]
@DanielePucci Same as my last comment, I still have to dedicate some time to understand how to properly test these classes. So far we didn't have any test to check possible regressions of these kind of featues.

File: src/sensors/src/ModelSensorsTransformers.cpp [hunk 1]
```diff
@@ -27,60 +27,152 @@ bool createReducedModelAndSensors(const Model& fullModel,
...
+            std::stringstream ss;
+            ss << "The processed FT sensor couldn't be cast as a joint sensor" << std::endl;
+            reportWarning("", "createReducedModelAndSensors", ss.str().c_str());
 
-            delete sensorCopy;
         }
     }
 
-    // Then all link sensors
+    // Then all link sensors (ACCELEROMETERs and GYROSCOPEs)
```
Comment (traversaro) [inline src/sensors/src/ModelSensorsTransformers.cpp]: There are several types of link sensors, and in the future we can add more, and the code is already generic w.r.t. to that. I would avoid embedding this kind of comments that can become eventually updated.

File: src/sensors/tests/ReducedModelWithFTUnitTest.cpp [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+#include <iDynTree/Sensors/ModelSensorsTransformers.h>
+#include <iDynTree/Model/ModelTransformers.h>
+#include <iDynTree/Model/Model.h>
+#include <iDynTree/ModelIO/URDFGenericSensorsImport.h>
+#include <iDynTree/ModelIO/URDFModelImport.h>
+#include "testModels.h"
+#include <iostream>
+
+// #include <thread>
+// #include <chrono>
```
Comment (traversaro) [inline src/sensors/tests/ReducedModelWithFTUnitTest.cpp]: Leftover?

File: src/sensors/tests/ReducedModelWithFTUnitTest.cpp [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+{
+    cout << "Sensor Name:\t" << ft.getName() << endl;
+    cout << "Is Valid: \t"   << ft.isValid() << endl;
+    cout << "First Link:\t"  << ft.getFirstLinkName() << endl;
+    cout << "Second Link:\t" << ft.getSecondLinkName() << endl;
+}
+
+int main()
+{
+    // this_thread::sleep_for(chrono::seconds(5));
```
Comment (traversaro) [inline src/sensors/tests/ReducedModelWithFTUnitTest.cpp]: Leftover?

**Comment by traversaro** [comment]
@diegoferigo There seems to be a memory problem in the tests, see https://travis-ci.org/robotology/idyntree/jobs/316335640 .

**Comment by diegoferigo** [comment]
I forgot a `delete` in the test unit and in my setup valgrind tests were disabled. Now should work.

File: src/sensors/src/ModelSensorsTransformers.cpp (hunk 1 above)
Comment (diegoferigo) [reply src/sensors/src/ModelSensorsTransformers.cpp]: Ok

File: src/sensors/tests/ReducedModelWithFTUnitTest.cpp (hunk 2 above)
Comment (diegoferigo) [reply src/sensors/tests/ReducedModelWithFTUnitTest.cpp]: Yep, debugging leftovers.

File: src/sensors/tests/ReducedModelWithFTUnitTest.cpp (hunk 3 above)
Comment (diegoferigo) [reply src/sensors/tests/ReducedModelWithFTUnitTest.cpp]: Yes

**Comment by diegoferigo** [comment]
OSX Travis' Build timed out but basically succeeded. Merging.

**Pull request merged**